integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/IcebergHandler.java

```diff
@@ -299,6 +306,12 @@ private String getCatalogType(Map<String, String> catalogConf) {
           "Default the catalog type to 'glue' because the catalog impl is {}",
           catalogConf.get(CATALOG_IMPL));
       return "glue";
+    } else if (catalogConf.containsKey(CATALOG_IMPL)
+        && catalogConf.get(CATALOG_IMPL).endsWith("RESTCatalog")) {
```

Comment:
Any reason we're not doing exact string matching but endsWith? 
We can potentially capture something we do not want here, for example `DeltaRESTCatalog` when we want `IcebergRESTCatalog`.

Reply:
I took same approach as with glue three lines above. The scenario you describe is impossible bcz is it's `IcebergHandler` and its methods are called only for `org.apache.iceberg.spark.SparkCatalog`. `CATALOG_IMPL` is canonical name, so perhaps the intention was to handle same implementation in different packages. Leaving this as is for now.